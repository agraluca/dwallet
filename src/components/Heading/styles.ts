import styled, { css } from "styled-components";

import { HeadingProps } from ".";

export const Wrapper = styled("h1").attrs<HeadingProps>(({ level }) => ({
  as: `h${level}`,
}))<HeadingProps>`
  ${({ theme, color }) => css`
    color: ${theme.colors[color!]};
  `}
`;
