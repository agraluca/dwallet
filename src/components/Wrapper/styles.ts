import styled, { css } from "styled-components";

export const Wrapper = styled.main`
  ${({ theme }) => css`
    background-color: ${theme.colors.black};
    width: 100%;
    height: 100%;
    position: relative;
  `}
`;

export const WaveImg = styled.img`
  position: absolute;
  bottom: 0;
  width: 100%;
`;

export const Content = styled.section`
  position: absolute;
  max-width: ${({ theme }) => theme.grid.container};
  margin: 0 auto;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
`;
