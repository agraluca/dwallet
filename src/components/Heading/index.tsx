import * as S from "./styles";

export type HeadingProps = {
  children: React.ReactNode;
  color?: "white" | "black";
  level?: 1 | 2 | 3 | 4 | 5 | 6;
  className?: string;
};

function Heading({
  children,
  color = "white",
  level = 1,
  className,
}: HeadingProps) {
  return (
    <S.Wrapper level={level} color={color} className={className}>
      {children}
    </S.Wrapper>
  );
}

export default Heading;
