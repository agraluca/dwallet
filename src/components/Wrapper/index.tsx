import { ReactNode } from "react";
import * as S from "./styles";

interface WrapperProps {
  children: ReactNode;
}

function Wrapper({ children }: WrapperProps) {
  return (
    <S.Wrapper>
      <S.Content>{children}</S.Content>
      <S.WaveImg src="/img/wave.svg" alt="wave" />
    </S.Wrapper>
  );
}

export default Wrapper;
