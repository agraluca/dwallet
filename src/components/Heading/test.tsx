import { screen } from "@testing-library/react";
import theme from "styles/theme";
import { renderWithTheme } from "utils/tests/helpers";

import Heading from ".";

describe("<Heading />", () => {
  it("should render a white heading by default", () => {
    renderWithTheme(<Heading>Zapt</Heading>);
    expect(screen.getByRole("heading", { name: /Zapt/i })).toHaveStyle({
      color: theme.colors.white,
    });
  });

  it("should render a black heading when color is passed", () => {
    renderWithTheme(<Heading color="black">Zapt</Heading>);
    expect(screen.getByRole("heading", { name: /Zapt/i })).toHaveStyle({
      color: theme.colors.black,
    });
  });
});
