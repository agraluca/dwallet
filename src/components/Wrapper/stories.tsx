import { Story, Meta } from "@storybook/react/types-6-0";

import Wrapper from ".";

export default {
  title: "Wrapper",
  component: Wrapper,
} as Meta;

export const Basic: Story = () => <Wrapper>test </Wrapper>;
