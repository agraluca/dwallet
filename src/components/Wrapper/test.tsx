import { screen } from "@testing-library/react";
import Heading from "components/Heading";
import { renderWithTheme } from "utils/tests/helpers";

import Wrapper from ".";

describe("<Wrapper />", () => {
  it("should render the heading", () => {
    const { container } = renderWithTheme(
      <Wrapper>
        <Heading level={1}>Wrapper</Heading>
      </Wrapper>
    );

    expect(
      screen.getByRole("heading", { name: /Wrapper/i })
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
