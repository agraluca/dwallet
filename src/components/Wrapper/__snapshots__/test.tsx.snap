// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Wrapper /> should render the heading 1`] = `
.c2 {
  color: #FAFAFA;
}

.c0 {
  background-color: #242423;
  position: relative;
  width: 100%;
  height: 100%;
}

.c3 {
  position: fixed;
  bottom: 0;
  width: 100%;
  pointer-events: none;
  z-index: -1;
}

.c1 {
  position: absolute;
  max-width: 126rem;
  min-width: 30rem;
  margin: 0 auto;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
}

<section
  class="c0"
>
  <section
    class="c1"
  >
    <h1
      class="c2"
      color="white"
    >
      Wrapper
    </h1>
  </section>
  <img
    alt="wave"
    class="c3 wave"
    src="/img/wave.svg"
  />
</section>
`;
